### PR TITLE
API Cleanup

### PR DESCRIPTION
**Pull request by shiftylogic**
* Normalizing argument order between Parsing / Formatting.
* Slight tweaks to formatting public surface.

### CODE REVIEW

**Comment by shiftylogic** [comment]
@ahsonkhan @KrzysztofCwalina @botaberg

File: src/System.Text.Primitives/System/Text/Formatting/PrimitiveFormatter.cs [hunk 1]
```diff
@@ -15,39 +15,39 @@ namespace System.Text
     public static partial class PrimitiveFormatter
     {
         #region Integers
-        public static bool TryFormat(this byte value, Span<byte> buffer, out int bytesWritten, TextFormat format, EncodingData encoding)
+        public static bool TryFormat(this byte value, Span<byte> buffer, out int bytesWritten, TextFormat format = default(TextFormat), EncodingData encoding = default(EncodingData))
```
Comment (KrzysztofCwalina) [inline src/System.Text.Primitives/System/Text/Formatting/PrimitiveFormatter.cs]: How did you chose between the two alternative orderings we could use? i.e. why do you think format followed by encoding is better than the other way around?

File: src/System.Text.Primitives/System/Text/Parsing/Signed.cs [hunk 2]
```diff
@@ -1,4 +1,10 @@
 ﻿
+
+
+
+
+
+
```
Comment (KrzysztofCwalina) [inline src/System.Text.Primitives/System/Text/Parsing/Signed.cs]: Why do we have these empty lines?

File: src/System.Text.Primitives/System/Text/Parsing/Unsigned.cs [hunk 3]
```diff
@@ -1,4 +1,10 @@
 ﻿
+
+
+
+
+
+
```
Comment (KrzysztofCwalina) [inline src/System.Text.Primitives/System/Text/Parsing/Unsigned.cs]: same

File: src/System.Text.Primitives/System/Text/Formatting/PrimitiveFormatter.cs (hunk 1 above)
Comment (shiftylogic) [inline src/System.Text.Primitives/System/Text/Formatting/PrimitiveFormatter.cs]: I suspect the case of passing an EncodingData is going to be much less frequent than formatting information. So, I'd argue to put the more commonly used parameter first.

I could be swayed with a better argument.

Comment (KrzysztofCwalina) [inline src/System.Text.Primitives/System/Text/Formatting/PrimitiveFormatter.cs]: Sounds good. Thanks.

File: src/System.Text.Primitives/System/Text/Parsing/Signed.cs (hunk 2 above)
Comment (shiftylogic) [inline src/System.Text.Primitives/System/Text/Parsing/Signed.cs]: I suspect that is an artifact of the T4 templates. It is one of my dislikes of T4 templates. The template engine does a very poor job of handling spacing / newlines / etc.

File: src/System.Text.Primitives/System/Text/Parsing/Unsigned.cs (hunk 3 above)
Comment (shiftylogic) [inline src/System.Text.Primitives/System/Text/Parsing/Unsigned.cs]: I suspect that is an artifact of the T4 templates. It is one of my dislikes of T4 templates. The template engine does a very poor job of handling spacing / newlines / etc.

File: src/System.Text.Json.Dynamic/System/Text/Json/JsonDynamicObject.cs [hunk 4]
```diff
@@ -273,9 +273,9 @@ public bool TryFormat(Span<byte> buffer, out int written, TextFormat format, Enc
                 switch (_type)
                 {
                     case JsonReader.JsonValueType.String:
-                        return _value.TryFormatQuotedString(buffer, format, formattingData, out written);
+                        return _value.TryFormatQuotedString(buffer, out written, format, formattingData: formattingData);
                     case JsonReader.JsonValueType.Number:
-                        return _value.TryFormat(buffer, format, formattingData, out written);
+                        return _value.TryFormat(buffer, out written, format, formattingData: formattingData);
                     case JsonReader.JsonValueType.Object:
```
Comment (ahsonkhan) [inline src/System.Text.Json.Dynamic/System/Text/Json/JsonDynamicObject.cs]: I thought the guideline was to have out parameters be at the end. Is that not the case?

File: src/System.Text.Primitives/System/Text/Formatting/PrimitiveFormatter.cs [hunk 5]
```diff
@@ -15,39 +15,39 @@ namespace System.Text
...
         {
             return IntegerFormatter.TryFormatUInt64(value, 1, buffer, out bytesWritten, format, encoding);
         }
 
-        public static bool TryFormat(this sbyte value, Span<byte> buffer, out int bytesWritten, TextFormat format, EncodingData encoding)
+        public static bool TryFormat(this sbyte value, Span<byte> buffer, out int bytesWritten, TextFormat format = default(TextFormat), EncodingData encoding = default(EncodingData))
         {
```
Comment (ahsonkhan) [inline src/System.Text.Primitives/System/Text/Formatting/PrimitiveFormatter.cs]: I thought the guideline was to have out parameters be at the end. Is that not the case? On the other hand, I also thought that default parameters go in the end. Is this the recommended approach when you have both?

Comment (KrzysztofCwalina) [inline src/System.Text.Primitives/System/Text/Formatting/PrimitiveFormatter.cs]: This is the official .NET guideline, but it means the parameters could not be defaulted. The guidelines were developed before we allowed default parameters in public APIs, and now these two guidelines (ok to use default and out parameters being last) seem not compatible with each other.

@terrajobst, we should talk about the guidelines at some point and see if we need to tweak them. But for now, I think this change is ok, and will give us more experience when we talk about the guidelines and how to adjust them.

**Pull request merged**